sklearn/ensemble/gradient_boosting_cv.py

```diff
@@ -0,0 +1,825 @@
+        self.max_depth = max_depth
+        self.init = init
+        self.random_state = random_state
+        self.max_features = max_features
+        self.max_leaf_nodes = max_leaf_nodes
+        self.presort = presort
+
+    def fit(self, X, y):
+        """Run fit with all sets of parameters till convergence.
+
+        X : array-like, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples
+            and n_features is the number of features.
+        y : array-like, shape = [n_samples]
+            Target values (integers in classification, real numbers in
+            regression)
+            For classification, labels must correspond to classes.
+        """
+
+        params = self._get_params()
+        X, y = check_X_y(X, y, dtype=DTYPE)
+
+        cv = check_cv(self.cv, y)
+        param_iter = ParameterGrid(params)
+
+        parallel = Parallel(n_jobs=self.n_jobs, verbose=self.verbose,
+                            backend='threading')
+
+        out = parallel(delayed(_fit_single_param)
+                       (self._estimator_class, X, y, train, validation, params,
+                        self.n_iter_no_change, self.max_iterations,
+                        self.scoring, self.score_precision, self.random_state)
+                       for train, validation in cv.split(X)
+                       for params in param_iter)
+
+        n_splits = int(len(out) / len(param_iter))
+
+        grid_scores = []
+        i = 0
+        best_mean = -np.inf
+        for params in param_iter:
+
+            score_list = []
+            n_est_list = []
+            for idx in range(n_splits):
+                score, n_estimators = out[i]
+                n_est_list.append(n_estimators)
+                score_list.append(score)
+                i += 1
+
+            scores = np.array(score_list)
+            mean = np.mean(scores)
+            params['n_estimators'] = int(np.mean(n_est_list))
+            score_tuple = _CVScoreTuple(params, mean, scores)
+            grid_scores.append(_CVScoreTuple(params, mean, scores))
+
+            if mean > best_mean:
+                best_tuple = score_tuple
+                best_params = params
+
+        self.grid_scores_ = grid_scores
+        self.best_score_tuple_ = best_tuple
+        self.best_params_ = best_params
+
+        if self.refit:
+            self.best_params_['random_state'] = self.random_state
+            gb = self._estimator_class(**self.best_params_)
+            gb.fit(X, y)
+            self.best_estimator_ = gb
+
+        return self
+
+    def predict(self, X):
+        """Call ``predict`` on the best estimator.
+
+        Parameters
+        ----------
+        X : array-like of shape = [n_samples, n_features]
+            The input samples.
+        Returns
+        -------
+        y: array of shape = ["n_samples]
+            The predicted values.
+        """
+        return self.best_estimator_.predict(X)
```

Comment:
Thanks for the catch. It used to be like how it was for `GridSearchCV`. #7234 fixed that. I'll make a similar fix here...